regparser/builder.py

```diff
@@ -237,7 +238,7 @@ def tree_and_builder(filename, title, checkpoint_path=None, doc_number=None):
     else:
         raise ValueError("Building from text input is no longer supported")
 
-    for preprocessor in all_preprocs.ALL:
+    for preprocessor in class_paths_to_classes(settings.PREPROCESSORS):
```

Comment:
Naive question, since I haven't dug into the preprocessors much, but do we need the indirection of instantiating a class with no arguments and calling a method--could `PREPROCESSORS` just be a sequence of callables? In the same vein, what's the benefit of defining an ABC that declares a single method as abstract, as opposed to using plain functions--or, if we want to bundle related functions together, a singleton class that defines `__call__`?

Pedantically, if a class combines instance state with behavior, classes may be unnecessary complications when there's no instance state.

/troll